src/LibraryPage.tsx

```diff
@@ -162,10 +189,32 @@ const LibraryPage: React.FC = () => {
         <TableColumn col="title">Title</TableColumn>
         <TableColumn col="author">Author</TableColumn>
         <TableColumn col="summary">Summary</TableColumn>
-        <Button onClick={() => setModalOpen(true)}>Edit</Button>
+        <BookEditButton
+          onEdit={(b) => {
+            setIsNewBook(false);
+            setModalOpen(true);
+            if (b) {
+              setSelectedBook(b);
+            }
+          }}
+        ></BookEditButton>
       </AutoTable>
     </PageLayout>
   );
 };
-
+interface editBookProps {
+  rowitem?: Book;
+  onEdit: (book: Book | undefined) => void;
+}
+const BookEditButton: React.FC<editBookProps> = ({ rowitem, onEdit }) => {
```

Comment:
I'd move this section (lines 205 and on) to a new file since this edit button is a separate component.